### ISSUE TITLE
Change Request: make --prune-supressions remove "empty" suppression file

### ISSUE THREAD

**Issue by karfau** [eslint/eslint#20335]
### ESLint version

9.35.0

### What problem do you want to solve?

When running eslint using the `--prune-suppressions` flag it produces an "empty" file `eslint-suppressions.json` containing just `{}`,
which, as far as I can see, has the same impact as if the file would not exist.

### What do you think is the correct solution?

When running eslint using the `--prune-suppressions` flag and there are no rules that need suppressing, remove the file `eslint-suppressions.json`

### Participation

- [x] I am willing to submit a pull request for this change.

### Additional comments

I have not contributed to eslint in the past, so some guidance where to start adding this behavior would be appreciated.

**Comment by nzakas** [comment]
Thanks for the suggestion. I'm not sure deleting the suppressions file would be an expected behavior for `--prune-suppressions`. What if instead we simply output a warning when the suppressions file is empty?

**Comment by karfau** [comment]
Hm, I would assume that something with `prune` in it's name cleans up after itself, in a way that it doesn't keep "empty" files, that do not have an impact on running eslint.

One reason why I'm saying this is, that when there are no suppressions and I use this flag it also **creates** that "empty" file only containing an empty object. This is something that I really didn't expect a `prune` command to cause...

So my argument is:
if having a file only containeing `{}` has the same effect as not having the file when running eslint, it should be good enough for eslint to drop the file:

No `eslint-suppressions.json` file, means I don't need to look at the content of the file to know that there are no suppressions. 
(In the same way that jest doesn't leave an empty snapshot file when removing the last `.toMatchSnapshot()` from a test and running the tests with `--updateSnapshots`.)

I would also totally fine with it being an option or even another CLI flag, whether or not to keep the "empty" file, so it isn't a breaking change for anybody.

But I'm not sure how "simply outputting a warning" and leaving it to the users to clean up files (spoiler: they are confused and don't do it) solves the issue. 

(The warning that is printed when supressions need to be pruned is not even recognizable when the usual eslint warning output fills the screen... I think this whole topic would actually benefit from being a rule that can be configured like any other rule and would lead to regular errors/warnings like other rules, but I assume this topic is for a different issue)

**Comment by nzakas** [comment]
Thanks for the feedback. We modeled the current behavior after the behavior that already existed in other linters. I'm not sure I agree that the term "prune" means "delete the file if there aren't any suppressions." The principle of least surprise indicates that we don't take actions that are side effects of the requested action.

I'm also not sure it makes sense to remove the suppressions file, in general. If you are using suppressions, then constantly deleting and re-creating the file creates unnecessary noise in source control. 

The most I'm willing to do is add a warning, but we can see what others think. @eslint/eslint-team

**Comment by DMartens** [comment]
I agree with nzakas point on the principle of least surprise and no hidden side effects.
Additionally the empty `eslint-suggestions.json` file shows that ESLint was previously run with suppressions enabled. If the file is missing but is expected to be present, one could suspect a bug (the CLI parameter may be specified elsewhere e.g. by specifying the lint script in the `package.json`).

**Comment by karfau** [comment]
As I said, if there is some config that alliws to explicitly opt in to deleting empty files, it would be expected, right?

And I was still not able to understand what the "warning" would help with, I never requested to add that, and if the "empty" file is what you expect to be there, why would you want to warn about anything?

**Comment by JoshuaKGoldberg** [comment]
I agree with the team's comments that a warning, rather than explicit file deletion, would be preferable in general. +1 to logging just a warning.

**Comment by nyarthan** [comment]
Maybe removing the file could be considered unexpected, but I certainly would not expect it to _create_ an empty file, if there are no suppressions.

**Comment by mdjermanovic** [comment]
I agree with @nzakas that automatically removing the suppressions file would be surprising. User can review the diff after running `--prune-suppressions` and remove the file manually if that's desired.

Also, when [`--suppressions-location`](https://eslint.org/docs/latest/use/command-line-interface#--suppressions-location) is used, any command would fail if the specified suppressions file doesn't exist. So, after removing the suppressions file, user would most likely need to update their scripts that run eslint to remove `--suppressions-location`, which is another reason not to remove the file automatically.

Overall, I think the current behavior (including creating an empty file) is correct and there's no need for any changes. If you're using the suppression feature, there should be a suppressions file. I don't think eslint should emit a warning either.

**Comment by karfau** [comment]
For anybody else looking for a workaround in the direction I requested originally: we added two scripts

`lint:prune` to run with `--prune-supressions`
And
```
   "postlint:prune": "find . -type d -name node_modules -prune -false -o -type f -name eslint-suppressions.json -exec sh -c 'jq -e \"length == 0\" \"$1\" >/dev/null 2>&1 && rm \"$1\"' _ {} \\;"
```
